WePeiYang/job/src/main/java/com/twt/service/job/home/JobFragment.kt

```diff
@@ -0,0 +1,145 @@
+    private lateinit var itemManager: ItemManager
+    private val homePresenterImp: JobHomePresenterImp = JobHomePresenterImp(this)
+    private var page: Int = 1
+    private var isLoad: Boolean = false
+
+    companion object {
+        fun newInstance(kind: String): JobFragment {
+            val args = Bundle()
+            args.putString(ARG_KIND, kind)
+            val jobFragment = JobFragment()
+            jobFragment.arguments = args
+            return jobFragment
+        }
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.apply {
+            kind = getString(ARG_KIND)
+        }
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
+        rootView = inflater.inflate(R.layout.job_fragment_home, container, false)
+        initView()
+        loadData()
+        refresh()
+        loadMoreData()
+        return rootView
+    }
+
+    private fun initView() {
+        recyclerView = rootView.findViewById(R.id.job_rv_homepage)
+        linearLayoutManager = LinearLayoutManager(activity, LinearLayoutManager.VERTICAL, false)
+        recyclerView.layoutManager = linearLayoutManager
+        swipeRefreshLayout = rootView.findViewById<SwipeRefreshLayout>(R.id.job_sr_refresh).apply {
+            setColorSchemeColors(Color.parseColor("#64a388"))
+        }
+    }
+
+    private fun refresh() {
+        swipeRefreshLayout.setOnRefreshListener {
+            page = 1
+            loadData()
+            swipeRefreshLayout.isRefreshing = false
+        }
+    }
+
+    private fun loadMoreData() {
+        recyclerView.addOnScrollListener(object : RecyclerView.OnScrollListener() {
+            var lastVisibleItem: Int = 0
+            var totalCount: Int = 0
+
+            override fun onScrolled(recyclerView: RecyclerView?, dx: Int, dy: Int) {
+                super.onScrolled(recyclerView, dx, dy)
+                lastVisibleItem = linearLayoutManager.findLastVisibleItemPosition()
+                totalCount = linearLayoutManager.itemCount
+                if (lastVisibleItem + 1 == totalCount && !isLoad) {
+                    isLoad = true
+                    ++page
+                    loadData()
+                }
+            }
+        })
+    }
+
+    private fun loadData() {
+        if (page > pagesOfMsg) {
+            itemManager.add(BottomItem())
+        } else {
+            homePresenterImp.getGeneral(kind, page)
+        }
+        isLoad = false
+    }
+
+    override fun showHomeFair(commonBean: List<HomeDataL>) {
+        recyclerView.withItems {
+            repeat(commonBean.size) { i ->
+                if (i == 0) fair(commonBean[i], true,this@JobFragment)
+                else fair(commonBean[i], false,this@JobFragment)
+            }
+        }
+        itemManager = (recyclerView.adapter as ItemAdapter).itemManager
```

Comment:
Follow the instruction of recDSL, the cast is avoidable.
```suggestion
        itemManager = ItemManager()
        recyclerView.adapter = ItemAdapter(itemManager)
```

Reply:
已修改